packages/node_modules/@webex/internal-plugin-devices/test/integration/spec/devices.js

```diff
@@ -0,0 +1,942 @@
+          });
+
+          it('uses the device\'s url property', () => {
+            assert.isTrue(devices.registered);
+
+            return devices.unregister()
+              .then(() =>
+                assert.calledWith(devices.request, sinon.match.has(
+                  'uri', url
+                )));
+          });
+
+          it('sends a DELETE request', () => {
+            assert.isTrue(devices.registered);
+
+            return devices.unregister()
+              .then(() =>
+                assert.calledWith(devices.request, sinon.match.has(
+                  'method', 'DELETE'
+                )));
+          });
+        });
+
+        describe('when the device unregistration request is successful', () => {
+          it('clears the device properties', () => devices.unregister()
+            .then(() => assert.isUndefined(devices.url)));
+        });
+
+        describe('when the device unregistration request fails', () => {
+          beforeEach('setup the \'request\' stub', () => {
+            devices.request = sinon.stub().rejects(
+              new WebexHttpError({
+                statusCode: 404,
+                options: {
+                  url: devices.url,
+                  headers: {
+                    trackingId: 'tid'
+                  }
+                }
+              })
+            );
+          });
+
+          it('returns a rejected promise', () => devices.unregister()
+            .then(() => assert.isTrue(false, 'resolved the promise'))
+            .catch(() => assert.isTrue(true, 'rejected the promise')));
+        });
+      });
+
+      describe('when the device is not registered', () => {
+        it('returns a resolved promise', () => devices.unregister()
+          .then(() => assert.isTrue(true, 'resolved the promise')));
+      });
+    });
+
+    describe('#waitForRegistration()', () => {
+      describe('when the device is registered', () => {
+        beforeEach('register the device', () => devices.register());
+
+        it('returns a resolved promise', () => devices.waitForRegistration()
+          .then(() => assert.isTrue(true, 'resolved the promise')));
+      });
+
+      describe('when the device is not registered', () => {
+        describe('when the device registers', () => {
+          beforeEach('trigger a pending device registration', () => {
+            devices.register();
+          });
+
+          it('returns a resolved promise once registered', () => {
+            assert.isFalse(devices.registered);
+
+            return devices.waitForRegistration()
+              .then(() => assert.isTrue(devices.registered));
+          });
+        });
+
+        describe('when the device does not register', () => {
+          it('returns a rejected promise', () => devices.waitForRegistration()
+            .then(() => assert.isTrue(false, 'resolved the promise'))
+            .catch(() => assert.isTrue(true, 'rejected the promise')));
+        });
+      });
+    });
+  });
+});
```

Comment:
Big test file for integration tests, this section needs reviewed in full.

Logic is:
```
describe: method
- describe: condition
- - describe: other condition
- - - it: expectation
- - - it: expectation
- - describe: other condition
- - - it: expectation
- - - it: expectation
- describe: condition
- - describe: other condition
- - - it: expectation
- - - it: expectation
::repeat::
```